worlds/terraria/Checks.py

```diff
@@ -0,0 +1,723 @@
+from BaseClasses import Item, Location
+import string
+import pkgutil
+
+
+class TerrariaItem(Item):
+    game = "Terraria"
+
+
+class TerrariaLocation(Location):
+    game = "Terraria"
+
+
+def add_token(
+    tokens,  #: list[tuple[int, int, str | int | None]],
+    token,  #: str | int,
+    token_index: int,
+):
+    if token == "":
+        return
+    if type(token) == str:
+        tokens.append((token_index, 0, token.rstrip()))
+    elif type(token) == int:
+        tokens.append((token_index, 1, token))
+
+
+IDENT = 0
+NUM = 1
+SEMI = 2
+HASH = 3
+AT = 4
+NOT = 5
+AND = 6
+OR = 7
+LPAREN = 8
+RPAREN = 9
+END_OF_LINE = 10
+
+CHAR_TO_TOKEN_ID = {
+    ";": SEMI,
+    "#": HASH,
+    "@": AT,
+    "~": NOT,
+    "&": AND,
+    "|": OR,
+    "(": LPAREN,
+    ")": RPAREN,
+}
+
+TOKEN_ID_TO_CHAR = {id: char for char, id in CHAR_TO_TOKEN_ID.items()}
+
+
+def tokens(rule: str):  # -> list[tuple[int, int, str | int | None]]:
+    tokens = []
```

Comment:
Variable name has the same name as your function. I'd recommend changing it for clarity.